mono/mini/interp/interp.c

```diff
@@ -4383,6 +4393,27 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, guint16 *st
 			++ip;
 			MINT_IN_BREAK;
 		}
+		MINT_IN_CASE(MINT_GETITEM_SPAN) {
+			guint8 *span = (guint8 *) sp [-2].data.p;
+			int index = sp [-1].data.i;
+			int element_size = (int) *(gint16 *) (ip + 1);
+			int offset_length = (int) *(gint16 *) (ip + 2);
+			int offset_pointer = (int) *(gint16 *) (ip + 3);
+			sp--;
+
+			if (!span)
+				THROW_EX (mono_get_exception_null_reference (), ip);
+
+			mono_array_size_t length = *(mono_array_size_t *) (span + offset_length);
+			if (index < 0 || index > length)
+				THROW_EX (mono_get_exception_index_out_of_range (), ip);
+
+			gpointer pointer = *(gpointer *)(span + offset_pointer);
+			sp [-1].data.p = (guint8 *) pointer + index * element_size;
```

Comment:
@lewurm it wouldn't hurt to cast here to avoid multiplication overflow

Reply:
This could use mono_array_addr_with_size_fast().